### PR TITLE
Mingshan/quantized avgpool + quantizedConcat

### PR DESCRIPTION
**Pull request by mingshan-wang**
This PR added QuantizedAvgPool, QuantizedConcat and QuantizedConcatV2. 

Verified with running quantized inception3 model and Quantized ops are falling on NGraph as expected, and the accuracy matches the TFDO team.

### CODE REVIEW

File: src/ngraph_builder.cc [hunk 1]
```diff
@@ -436,6 +436,86 @@ static Status TranslateBinaryOp(
...
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map, std::string pooling_name) {
+  bool is_quantizedAvgPool = pooling_name == "QuantizedAvgPool" ? true : false;
+  bool is_quantizedMaxPool = pooling_name == "QuantizedMaxPool" ? true : false;
+
+  // Doesn't  compile
+  // bool is_quantizedAvgPool =
+  // std::is_same<T,ng::builder::ScaledQuantizedAvgPool>::value;
+  // bool is_quantizedMaxPool =
+  // std::is_same<T,ng::builder::ScaledQuantizedMaxPool>::value;
```
Comment (sayantan-nervana) [inline src/ngraph_builder.cc]: we can remove these comments. `ScaledQuantizedAvgPool`, `ScaledQuantizedMaxPool ` are builder functions (and not classes like ng nodes), so we can't template them

File: src/ngraph_builder.cc [hunk 2]
```diff
@@ -436,6 +436,86 @@ static Status TranslateBinaryOp(
       });
 }
 
+// Helper function for translating QuantizedAvgPool and QuantizedMaxPool
+static Status TranslateQuantizedPoolOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map, std::string pooling_name) {
+  bool is_quantizedAvgPool = pooling_name == "QuantizedAvgPool" ? true : false;
+  bool is_quantizedMaxPool = pooling_name == "QuantizedMaxPool" ? true : false;
```
Comment (sayantan-nervana) [inline src/ngraph_builder.cc]:
```suggestion
  bool is_quantizedMaxPool = pooling_name == "QuantizedMaxPool";
```

File: src/ngraph_builder.cc [hunk 3]
```diff
@@ -436,6 +436,86 @@ static Status TranslateBinaryOp(
       });
 }
 
+// Helper function for translating QuantizedAvgPool and QuantizedMaxPool
+static Status TranslateQuantizedPoolOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map, std::string pooling_name) {
+  bool is_quantizedAvgPool = pooling_name == "QuantizedAvgPool" ? true : false;
```
Comment (sayantan-nervana) [inline src/ngraph_builder.cc]:
```suggestion
  bool is_quantizedAvgPool = pooling_name == "QuantizedAvgPool";
```

**Comment by mingshan-wang** [comment]
TESTNOW

**Comment by sayantan-nervana** [comment]
TESTNOW

**Comment by mingshan-wang** [comment]
TESTNOW

File: src/ngraph_builder.cc [hunk 4]
```diff
@@ -2422,6 +2496,129 @@ static Status TranslateQuantizeAndDequantizeV2Op(
...
+
+    auto min_node =
+        make_shared<ng::op::Constant>(ng::element::f32, ng::Shape{}, min_tmp);
+    auto max_node =
+        make_shared<ng::op::Constant>(ng::element::f32, ng::Shape{}, max_tmp);
+
+    ng_all_mins.push_back(std::make_shared<ngraph::op::Reshape>(
+        min_node, ngraph::AxisVector{}, ngraph::Shape{1}));
+    ng_all_maxs.push_back(std::make_shared<ngraph::op::Reshape>(
+        max_node, ngraph::AxisVector{}, ngraph::Shape{1}));
```
Comment (sayantan-nervana) [inline src/ngraph_builder.cc]: Can we avoid this extra reshape? (Bu supplying appropriate args in the `Constant` constructor above

File: test/test_array_ops.cpp [hunk 5]
```diff
@@ -516,6 +516,86 @@ TEST(ArrayOps, QuantizeAndDequantizeV2x8xtruexfalse) {
...
+  //   PrintTensorAllValues(i, 100);
+  // }
+
+  // cout << "NG outputs " << endl;
+  // for(auto i : ng_outputs){
+  //   PrintTensorAllValues(i, 100);
+  // }
+  opexecuter.RunTest();
+}  // end of test op QuantizedConcatVaryingMinMax
+
```
Comment (sayantan-nervana) [inline test/test_array_ops.cpp]: These 2 unit tests are for `QuantizedConcat`. Is it possible to add an unit test for `QuantizedConcatV2` as well?

File: src/ngraph_builder.cc (hunk 4 above)
Comment (mingshan-wang) [reply src/ngraph_builder.cc]: Tried that, using Constant with `ng::Shape{1}`, but I got error doing that way.

File: test/test_array_ops.cpp (hunk 5 above)
Comment (mingshan-wang) [reply test/test_array_ops.cpp]: There is no TF QuantizedConcatV2 API, I got error sth like TF doesn't have this op registered.

**Comment by mingshan-wang** [comment]
TESTNOW

**Comment by mingshan-wang** [comment]
TESTNOW

**Pull request merged**